### PR TITLE
Updated react-native-dotenv

### PR DESCRIPTION
**Pull request by hmajid2301**
Updated react-native-dotenv tried to use syntax like this #34600. However the library throws an
error https://github.com/zetachang/react-native-dotenv/issues/5 as discussed in this issue.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/zetachang/react-native-dotenv/issues/5
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/react-native-dotenv/index.d.ts [hunk 1]
```diff
@@ -4,9 +4,5 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 3.0
 
-interface Env {
-    [name: string]: string;
-}
-
-declare const env: Env;
-export = env;
+declare const _a: any;
```
Comment (PranavSenthilnathan) [inline types/react-native-dotenv/index.d.ts]: I'm not quite sure what this change is intending to solve. Could you add a test case to give an example of the problem and how this fixes it?

**Review by PranavSenthilnathan — CHANGES_REQUESTED** [review]

File: types/react-native-dotenv/index.d.ts (hunk 1 above)
Comment (hmajid2301) [reply types/react-native-dotenv/index.d.ts]: In  #34600, it was recommended to import the library `import env from "react-native-dotenv"`.
However if you try this it throw an error, as per this issue  zetachang/react-native-dotenv#5. 
So thats what I'm trying to resolve.

**Comment by hmajid2301** [comment]
Changes made.

**Comment by hmajid2301** [comment]
@PranavSenthilnathan not sure if you saw my comments.

In #34600, it was recommended to import the library import env from "react-native-dotenv".
However if you try this it throw an error, as per this issue zetachang/react-native-dotenv#5.
So thats what I'm trying to resolve.

**Comment by PranavSenthilnathan** [comment]
I think there are two parts to what you want to solve.

1) Allowing arbitrary property access for a module. This isn't possible: https://stackoverflow.com/a/31729755

2) Disallow importing of the entire module (to force users to extract the properties in the import statement). Again, this isn't possible since the import syntax always allows importing the whole module.

I'm not sure how this plugin works internally but as I understand it does some rewriting of the import statement itself, so that might be the mechanism it uses to give an error when people import the wrong things. But from the original JavaScript/TypeScript language syntax, this isn't something that's supported.

@sandersn

**Comment by hmajid2301** [comment]
To be honest I just wanted to hide the compiler warning by having some definitions for the library.

**Comment by chrisdrackett** [comment]
@hmajid2301 have you gotten around the "@types/react-native-dotenv"' has no exported member" issues somehow? are you just adding ignores?

**Comment by hmajid2301** [comment]
When I was using this package I was using Expo, since then I've ejected and I am using `react-native-config`.

**Pull request closed without merging**